views/js/runner/plugins/navigation/previous.js

```diff
@@ -66,15 +79,46 @@ define([
             }));
 
             //attach behavior
-            this.$element.on('click', function(e){
-                e.preventDefault();
-                if(self.getState('enabled') !== false){
-                    self.disable();
+            function doPrevious(previousItemWarning) {
+                var context = testRunner.getTestContext();
 
-                    testRunner.previous();
+                previousItemWarning = previousItemWarning || false;
+
+                function enable() {
+                    testRunner.trigger('enablenav enabletools');
                 }
+
+                if(self.getState('enabled') !== false){
+                    testRunner.trigger('disablenav disabletools');
+
+                    if (previousItemWarning && context.remainingAttempts !== -1) {
+                        testRunner.trigger(
+                            'confirm.previous',
+                            __('You are about to go to the previous item. Click OK to continue and go to the previous item.'),
+                            testRunner.previous, // if the test taker accept
+                            enable  // if the test taker refuse
+                        );
+
+                    } else {
+                        testRunner.previous();
+                    }
+                }
+            }
+
+            this.$element.on('click', function(e){
+                e.preventDefault();
+                testRunner.trigger('nav-previous');
             });
 
+            if(testConfig && testConfig.allowShortcuts){
+                shortcut.add(namespaceHelper.namespaceAll(pluginShortcuts.toggle, this.getName(), true), function(e) {
+                    if (canDoPrevious() && self.getState('enabled') === true) {
+                        e.preventDefault();
```

Comment:
Same as above. This instruction is already invoked by default by the shortcut helper. At least you should use the `prevent` option to be sure the event is prevented.